includes/Core/McpAdapter.php

```diff
@@ -54,146 +33,51 @@ class McpAdapter {
 	private array $servers = array();
 
 	/**
-	 * The has triggered init flag.
-	 *
-	 * @var bool
-	 */
-	private bool $has_triggered_init = false;
-
-	/**
-	 * Constructor
-	 */
-	private function __construct() {
-		if ( self::$initialized || self::$initialization_failed ) {
-			return;
-		}
-
-		if ( ! $this->check_dependencies() ) {
-			self::$initialization_failed = true;
-			return;
-		}
-		add_action( 'rest_api_init', array( $this, 'mcp_adapter_init' ), 20000 );
-		self::$initialized = true;
-	}
```

Comment:
Moving the `add_action()` to `::instance()` allows us to clean out the constructor entirely, meaning we can test and use without needing additional class properties for lifecycle state.